shell/shared/grain.js

```diff
@@ -1300,6 +1304,7 @@ Router.map(function () {
     },
 
     onStop: function () {
+      this.state.set("beforeActionHookRan", undefined);
```

Comment:
Is there idea on this line to have onStop() set this value to (basically) false? If so it's arguably cleaner to use a real boolean of `true` and `false`, and then where you call `this.state.get()` check if the return value is truthy.

I can live with this as-is, but mostly just wanted to say that intentionally using `undefined` as a value in Javascript weirds me out a little. (-: